src/contracts/signing-key-events-cache/leveldb/leveldb.service.ts

```diff
@@ -199,6 +205,28 @@ export class LevelDBService {
     await this.db.batch(ops);
   }
 
+  private validateHeader(
```

Comment:
Perhaps we should think about some kind of library for schema validation? Like zod?

Reply:
I am not sure that we should install a library for just one check. If this check is enough, I will not delegate it to a library.